src/cr/cube/cube.py

```diff
@@ -773,6 +793,50 @@ def _flat_values(self):
         ).flatten()
 
 
+class _OverlapMeasure(_BaseMeasure):
+    """Statistical overlap values from a cube-response."""
+
+    @lazyproperty
+    def missing_count(self):
+        """Numeric value representing count of missing rows in response."""
+        return self._cube_dict["result"]["measures"]["overlap"].get("n_missing", 0)
+
+    @lazyproperty
+    def _measure_payload(self):
+        return self._cube_dict["result"].get("measures", {}).get("overlap")
+
+    @lazyproperty
+    def _flat_values(self):
+        """Optional 1D np.ndarray of np.float64 overlap values as found in cube response.
+
+        Overlap data may include missing items represented by a dict like
+        {'?': -1} in the cube response. These are replaced by np.nan in the
+        returned value.
+        """
+        if self._measure_payload is None:
+            return None
+        return np.array(
+            tuple(
+                np.nan if type(x) is dict else x for x in self._measure_payload["data"]
+            ),
+            dtype=np.float64,
+        ).flatten()
+
+    @lazyproperty
+    def _shape(self):
+        n_subvars = len(self._measure_payload["metadata"]["type"]["subvariables"])
+        return self._all_dimensions.shape + (n_subvars,)
+
+    @lazyproperty
+    def raw_cube_array(self):
+        if self._flat_values is None:
+            return None
+        raw_cube_array = self._flat_values.reshape(self._shape)
+        # ---must be read-only to avoid hard-to-find bugs---
+        raw_cube_array.flags.writeable = False
+        return raw_cube_array
```

Comment:
I don't see the reason of having this code fragment duplicated.
I would suggest to create a shape property in `_BaseMeasure`:
```python3
@lazyproperty
def _shape(self):
    """ ... """
    return self._all_dimensions.shape
```
and then override this shape in the `Overlap` measure and get rid of the `raw_cube_array` that is basically the same code of the `BaseMeasure`

Reply:
Done